app/models/event.rb

```
@@ -286,6 +293,27 @@ def self.by_event_start_date
     order(:event_start_date)
   end
 
+  ##
+  # Loads PSC data for an ActiveRecord::Relation of events.
+  #
+  # This method forces evaluation of a relation, so you can only use it at the
+  # end of a scope chain.
+  #
+  # This method does not implement any special error handling for PSC queries.
+  #
+  # @param [PatientStudyCalendar] psc a PatientStudyCalendar object
+  # @return Array<Event>
```

This isn't always true: `PatientStudyCalendar` currently swallows all exceptions and returns `nil` from calls that trigger such exceptions.  That needs to be fixed.